agent/consul/state/intention.go

```diff
@@ -853,6 +837,16 @@ func (s *Store) IntentionMatchOne(
 	tx := s.db.Txn(false)
 	defer tx.Abort()
 
+	return compatIntentionMatchOneTxn(tx, ws, entry, matchType)
+}
+
+func compatIntentionMatchOneTxn(
```

Comment:
Why prefix this with `compat` if you're already suffixing it with `Txn`?

Reply:
There's already an `intentionMatchOneTxn`. I first thought to rename that one to `legacyIntentionMatchOneTxn` since it does legacy things, but that also exists, and calls the former. So I named this one compat because it has the config/legacy split in it.

What do you think?

Reply:
:+1: